src/rtk/features/overviewOrg/overviewOrgSlice.ts

```diff
@@ -193,54 +160,36 @@ const overviewOrgSlice = createSlice({
       }
     },
     save: (state) => {
-      const list = mapToSoftDeletedOrgDto(state);
-      console.log(list);
+      console.log(state.overviewOrgs);
     },
-    softUndeleteAll: (state, action) => {
-      const undeletedOrgId = action.payload;
-      for (const org of state.softDeletedOverviewOrgs) {
-        if (org.id === undeletedOrgId) {
-          removeSoftDeletedOrgReference(state, undeletedOrgId);
-          break;
-        }
-      }
+    softRestoreAll: (state, action) => {
+      const restoredOrgId = action.payload;
 
-      setSoftDeleteState(state, undeletedOrgId, false);
+      setAllItemsToGivenSoftDeleteState(state, restoredOrgId, false);
     },
     softDeleteAll: (state, action) => {
       const selectedOrgId = action.payload;
-      for (const softDeletedOrg of state.softDeletedOverviewOrgs) {
-        if (softDeletedOrg.id === selectedOrgId) {
-          removeSoftDeletedOrgReference(state, selectedOrgId);
-        }
-      }
-      state.softDeletedOverviewOrgs.push(selectedOrgId);
 
-      setSoftDeleteState(state, selectedOrgId, true);
-    },
-    setIsEditable: (state, action) => {
-      state.overviewOrgIsEditable = action.payload;
+      setAllItemsToGivenSoftDeleteState(state, selectedOrgId, true);
     },
-    emptySoftDeletedList: (state) => {
+    emptySoftDeletedItems: (state) => {
       for (const org of state.overviewOrgs) {
         org.isAllSoftDeleted = false;
```

Comment:
Drop this line
It's repeated after the for-loop anyway

Reply:
I'll delete the line after the for-loop instead because of readability

Reply:
I usually prefer to do the opposite because of fault tolerance.

If something happens that interrupts the process before the for-loop is finished and everything has been set to softDelete = false, then this flag will be wrong. Because the deletion process didn't finish.
That's why I prefer to set such states only after you can guarantee their truthyness.

Idk, might be paranoia, but it's what I've learned from my cybernetics background

Reply:
Valid point